source/uwp/Renderer/lib/ActionHelpers.h

```diff
@@ -4,79 +4,104 @@
-                            _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderContext* renderContext,
-                            _In_ ABI::Windows::UI::Xaml::IUIElement* uiElement,
-                            bool supportsInteractivity,
-                            bool fullWidthTouchTarget,
-                            _COM_Outptr_ ABI::Windows::UI::Xaml::IUIElement** outUiElement);
-
-    HRESULT BuildActions(_In_ ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveCard* adaptiveCard,
-                         _In_ ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveActionElement*>* children,
-                         _In_ ABI::Windows::UI::Xaml::Controls::IPanel* bodyPanel,
-                         bool insertSeparator,
-                         _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderContext* renderContext,
-                         _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderArgs* renderArgs);
-
-    HRESULT BuildActionSetHelper(_In_opt_ ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveCard* adaptiveCard,
-                                 _In_opt_ ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveActionSet* adaptiveActionSet,
-                                 _In_ ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveActionElement*>* children,
-                                 _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderContext* renderContext,
-                                 _In_ ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderArgs* renderArgs,
-                                 _Outptr_ ABI::Windows::UI::Xaml::IUIElement** actionSetControl);
-
-    void CreateAppropriateButton(_In_ ABI::AdaptiveCards::ObjectModel::Uwp::IAdaptiveActionElement* action,
-                                 Microsoft::WRL::ComPtr<ABI::Windows::UI::Xaml::Controls::IButton>& button);
+    void HandleKeydownForInlineAction(winrt::Windows::UI::Xaml::Input::KeyRoutedEventArgs const& args,
+                                      winrt::AdaptiveCards::Rendering::Uwp::AdaptiveActionInvoker const& actionInvoker,
+                                      winrt::AdaptiveCards::ObjectModel::Uwp::IAdaptiveActionElement const& inlineAction);
+
+    winrt::Windows::UI::Xaml::UIElement
+    HandleInlineAction(winrt::AdaptiveCards::Rendering::Uwp::AdaptiveRenderContext const& renderContext,
+                       winrt::AdaptiveCards::Rendering::Uwp::AdaptiveRenderArgs const& renderArgs,
+                       winrt::Windows::UI::Xaml::UIElement const& textInputUIElement,
+                       winrt::Windows::UI::Xaml::UIElement const& textBoxParentContainer,
+                       bool isMultilineTextBox,
+                       winrt::AdaptiveCards::ObjectModel::Uwp::IAdaptiveActionElement const& inlineAction);
+
+    void WireButtonClickToAction(winrt::Windows::UI::Xaml::Controls::Button button,
```

Comment:
>Button [](http://example.com/codeflow?start=69&length=6)

stick with 'const&' everywhere

Reply:
Well not literally *everywhere*. We still want to take coroutine parameter by value, not reference.